raster/r.horizon/main.c

```diff
@@ -57,94 +59,86 @@ module by Jaro Hofierka
 #define DISTANCE1(x1, x2, y1, y2) \
     (sqrt((x1 - x2) * (x1 - x2) + (y1 - y2) * (y1 - y2)))
 
-FILE *fp;
-
 const double pihalf = M_PI * 0.5;
 const double twopi = M_PI * 2.;
 const double invEarth = 1. / EARTHRADIUS;
 const double deg2rad = M_PI / 180.;
 const double rad2deg = 180. / M_PI;
-const double minAngle = DEG;
-
-const char *elevin;
-const char *horizon = NULL;
-const char *mapset = NULL;
-const char *per;
-char *shad_filename;
-char *outfile;
 
-struct Cell_head cellhd;
-struct Key_value *in_proj_info, *in_unit_info;
 struct pj_info iproj, oproj, tproj;
+float **z, **z100, **horizon_raster;
+int ll_correction = FALSE;
 
-struct Cell_head new_cellhd;
-double bufferZone = 0., ebufferZone = 0., wbufferZone = 0., nbufferZone = 0.,
-       sbufferZone = 0.;
-
-int INPUT(void);
-int OUTGR(int numrows, int numcols);
+typedef struct {
+    double xg0, yg0;
+    double z_orig;
+    double coslatsq;
+    double maxlength;
+} OriginPoint;
+
+typedef struct {
+    double stepsinangle, stepcosangle;
+    double sinangle, cosangle;
+    double distsinangle, distcosangle;
+} OriginAngle;
+
+typedef struct {
+    double xx0, yy0;
+    int ip, jp;
+    int ip100, jp100;
+    double zp;
+} SearchPoint;
+
+typedef struct {
+    double tanh0;
+    double length;
+} HorizonProperties;
+
+typedef struct {
+    int n, m, m100, n100;
+    double stepx, stepy, stepxy;
+    double invstepx, invstepy;
+    double offsetx, offsety;
+    double distxy;
+    double xmin, xmax, ymin, ymax, zmax;
+} Geometry;
+
+typedef struct {
+    int degreeOutput;
+    int compassOutput;
+    double fixedMaxLength;
+    double start, end, step;
+    double single_direction;
+    const char *str_step;
+    const char *horizon_basename;
+} Settings;
+
+int INPUT(Geometry *geometry, const char *elevin);
+int OUTGR(const Settings *settings, char *shad_filename,
+          struct Cell_head *cellhd);
 double amax1(double, double);
-double amin1(double, double);
 int min(int, int);
```

Comment:
Would it be possible to replace `min()` with
https://github.com/OSGeo/grass/blob/4d2dac47893779757585605b70903dc2277ba23c/include/grass/gis.h#L154